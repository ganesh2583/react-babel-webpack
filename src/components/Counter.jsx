// @flow

import React, {useState} from "react";

const Counter = () => {
  const [count: number, incrementCountByOne] = useState<number>(0);
  return (
    <div id="counterId">
      Counter = {count}
      <button
        onClick={() => {
          incrementCountByOne(count + 1);
        }}
        id="incrementBtn"
      >
        Click to increment
      </button>
    </div>
  );
};

export default Counter;
