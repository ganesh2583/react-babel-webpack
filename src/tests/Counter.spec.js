import React from "react";
import renderer from "react-test-renderer";
import {shallow, configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Counter from "./../components/Counter";

configure({adapter: new Adapter()});

describe("Test Counter", () => {
  it("Test Counter Render on Snapshot", () => {
    const component = renderer.create(<Counter />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Test onClick increase the count", () => {
    const component = shallow(<Counter />);
    expect(component.find("#counterId").text()).toEqual(
      "Counter = 0Click to increment"
    );
    component
      .find("#incrementBtn")
      .props()
      .onClick();
    expect(component.find("#counterId").text()).toEqual(
      "Counter = 1Click to increment"
    );
  });
});
