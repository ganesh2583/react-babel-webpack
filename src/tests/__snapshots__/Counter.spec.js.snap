// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Counter Test Counter Render on Snapshot 1`] = `
<div
  id="counterId"
>
  Counter = 
  0
  <button
    id="incrementBtn"
    onClick={[Function]}
  >
    Click to increment
  </button>
</div>
`;

exports[`Test Counter Test onClick increase the count 1`] = `
<span>
  Counter = 
  0
  <button
    onClick={[Function]}
  >
    Click to increment
  </button>
</span>
`;
